src/main/java/com/algorand/algosdk/v2/client/algod/GetApplicationBoxByName.java

```diff
@@ -0,0 +1,82 @@
+package com.algorand.algosdk.v2.client.algod;
+
+import com.algorand.algosdk.v2.client.common.Client;
+import com.algorand.algosdk.v2.client.common.HttpMethod;
+import com.algorand.algosdk.v2.client.common.Query;
+import com.algorand.algosdk.v2.client.common.QueryData;
+import com.algorand.algosdk.v2.client.common.Response;
+import com.algorand.algosdk.v2.client.model.Box;
+
+
+/**
+ * Given an application ID and box name, it returns the box name and value (each
+ * base64 encoded). Box names must be in the goal app call arg encoding form
+ * 'encoding:value'. For ints, use the form 'int:1234'. For raw bytes, use the form
+ * 'b64:A=='. For printable strings, use the form 'str:hello'. For addresses, use
+ * the form 'addr:XYZ...'.
+ * /v2/applications/{application-id}/box
+ */
+public class GetApplicationBoxByName extends Query {
+
+    private Long applicationId;
+
+    /**
+     * @param applicationId An application identifier
+     */
+    public GetApplicationBoxByName(Client client, Long applicationId) {
+        super(client, new HttpMethod("get"));
+        this.applicationId = applicationId;
+    }
+
+    /**
+     * A box name, in the goal app call arg form 'encoding:value'. For ints, use the
+     * form 'int:1234'. For raw bytes, use the form 'b64:A=='. For printable strings,
+     * use the form 'str:hello'. For addresses, use the form 'addr:XYZ...'.
+     */
+    public GetApplicationBoxByName name(String name) {
```

Comment:
This pattern suffers from the same drawbacks I brought up in https://github.com/algorand/js-algorand-sdk/pull/604#discussion_r975688014

Though since this code is generated, I expect we don't want to take any action here at the moment.